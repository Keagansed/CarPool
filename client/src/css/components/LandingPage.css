body{
    background-color: #140D4F;
}

#imgLogo{
    margin-bottom: 2rem;
}

#btnLogin{
    margin-bottom: 1rem;
}

#btnLogin, #btnRegister{
    width: 15rem;
    
    border-radius: 2rem;
}