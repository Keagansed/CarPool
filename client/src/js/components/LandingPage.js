import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import LoginStore from './../stores/LoginStore';
import logo  from "./../../css/images/logo.png";
import fbIcon from "./../../css/images/fb_icon.png";
import googleIcon from "./../../css/images/google_icon.png";


class LandingPage extends Component{
    componentWillMount(){
        LoginStore.setRegistered(false);
    }

    render(){
        return(
            <div>

                <div className="container">
                    <div className="row">
                        <div className="col-md-12 my-4">
                            <img className="img-fluid d-block mx-auto" src={logo} id="imgLogo" alt="logo"/> 
                        </div>
                    </div>
                </div>

                <div className="text-center">
                    <div className="container">
                    <div className="row">
                        <div className="col-md-12">
                        <Link to={`/Login`} className="btn w-75 text-center btn-primary text-secondary roundCorners font-weight-medium" id="btnLogin">
                            <b>Login</b>
                        </Link>
                        </div>
                    </div>
                    <div className="row">
                        <div className="col-md-12">
                        <Link to={`/Register`} className="btn w-75 text-center btn-primary text-secondary roundCorners my-3 font-weight-medium" id="btnRegister">
                            <b>Register</b>
                        </Link>
                        </div>
                    </div>
                    </div>
                </div>

                <div className="py-2 text-center">
                    <div className="container">
                        <div className="row">
                            <div className="col-md-12">
                                <p className="" id="dividerOR">
                                    <strike>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strike>&nbsp; &nbsp; <b className="small font-weight-medium">OR</b> &nbsp; &nbsp;
                                    <strike>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;</strike>
                                </p>
                            </div>
                        </div>
                    </div>
                </div>

                <div className="p-0">
                    <div className="text-center container">
                        <div className="row text-center">
                            <div className="col-2-pr-5 col-sm-2 col-md-6 col-lg-2 col-xl-2 ml-auto">
                                <img className="img-fluid d-block mx-auto" src={fbIcon} id="btnFacebookLogin" alt="fbLogin"/> 
                            </div>
                            <div className="col-2-pl-5 col-sm-2 col-md-6 col-lg-2 col-xl-2 mr-auto">
                                <img className="img-fluid d-block mx-auto" src={googleIcon} id="btnGoogleLogin" alt="googleLogin"/> 
                            </div>
                        </div>
                    </div>
                </div>

            </div>
        );
    }
}

export default LandingPage;