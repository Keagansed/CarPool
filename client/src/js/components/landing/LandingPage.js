import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import LoginStore from './../../stores/LoginStore'
import logo  from "./../../../css/images/logo.png";


class LandingPage extends Component{
    componentWillMount(){
        LoginStore.setRegistered(false);
    }

    render(){
        return(
            <div className="vertical-center bg-purple">
                <div className="container-fluid">
                    <div className="row">
                        <img className="img-fluid d-block mx-auto mbottom-2rem" src={logo} alt="carpool_logo"/> 
                    </div>
                    <div className="row">
                        <Link to={`/Login`} className="btn btn-primary mx-auto width-15rem brad-2rem mbottom-1rem bg-aqua txt-purple fw-bold" id="btnLogin">
                            Login
                        </Link>
                    </div>
                    <div className="row">
                        <Link to={`/Register`} className="btn btn-primary mx-auto width-15rem brad-2rem bg-aqua txt-purple fw-bold" id="btnRegister">
                            Register
                        </Link>
                    </div>
                </div>
            </div>
        );
    }
}

export default LandingPage;