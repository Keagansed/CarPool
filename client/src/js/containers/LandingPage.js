// File Type: Component

import { Link } from 'react-router-dom';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import withStyles from '@material-ui/core/styles/withStyles';

import LoginStore from '../stores/LoginStore';
import logo from "../../css/images/logo.png";

//Define the spefic styles for this page
const styles = theme => ({
    layout: {
        width: 'auto',
        display: 'block', // Fix IE11 issue.
        marginLeft: theme.spacing.unit * 3,
        marginRight: theme.spacing.unit * 3,
        [theme.breakpoints.up(400 + theme.spacing.unit * 3 * 2)]: {
            width: 400,
            marginLeft: 'auto',
            marginRight: 'auto',
        },
    },
    paper: {
        paddingTop: theme.spacing.unit * 8,
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        backgroundColor: 'transparent',
        boxShadow: 'none',
        padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 3}px ${theme.spacing.unit * 3}px`,
    },
    avatar: {
        margin: theme.spacing.unit,
        height: 128,
        width: 128,
    },
    form: {
        width: '100%', // Fix IE11 issue.
    },
    submit: {
        marginTop: theme.spacing.unit * 3,
    },
});

/*
* Purpose: Landing page compenent is the first page the user sees when opening the app
*/
class LandingPage extends Component {
    componentWillMount() {
        LoginStore.setRegistered(false);
    }

    render() {
        const { classes } = this.props;

        return (
            <React.Fragment>
                <CssBaseline />
                <main className={classes.layout}>
                    <Paper className={classes.paper}>
                        <Avatar src={logo} align='center' className={classes.avatar} />
                        <Typography align='center' variant="headline">The Iminsys Carpool Platform</Typography>
                        <form className={classes.form}>
                            <Link to={`/Register`} style={{ textDecoration: 'none' }}>
                                <Button
                                    type="submit"
                                    fullWidth
                                    variant="raised"
                                    color="primary"
                                    className={classes.submit}
                                >
                                    Register
                                </Button>
                            </Link>
                            <Link to={`/Login`} style={{ textDecoration: 'none' }}>
                                <Button
                                    type="submit"
                                    fullWidth
                                    variant="raised"
                                    color="primary"
                                    className={classes.submit}
                                >
                                    Login
                                </Button>
                            </Link>
                        </form>
                    </Paper>
                </main>
            </React.Fragment>
        );
    }
}

LandingPage.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(LandingPage);