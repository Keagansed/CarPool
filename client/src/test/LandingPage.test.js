import React from 'react';
import LandingPage from '../js/components/LandingPage';
import { shallow } from 'enzyme';
// let renderer = require('react-test-renderer');

describe('Landing Page Component', () => {
    it('renders correctly', () => {
        // const tree = renderer.create(<LandingPage />).toJSON();
        // expect(tree).toMatchSnapshot();

        expect(shallow(<LandingPage />).exists(<div className="container"></div>)).toBe(true);
    });

});