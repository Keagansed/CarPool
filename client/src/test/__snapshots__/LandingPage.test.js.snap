// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Component captures snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12 my-4"
      >
        <img
          alt="logo"
          className="img-fluid d-block mx-auto"
          id="imgLogo"
          src="logo.png"
        />
      </div>
    </div>
  </div>
  <div
    className="text-center"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <a
            className="btn w-75 text-center btn-primary text-secondary roundCorners font-weight-medium"
            href="/Login"
            id="btnLogin"
            onClick={[Function]}
          >
            <b>
              Login
            </b>
          </a>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <a
            className="btn w-75 text-center btn-primary text-secondary roundCorners my-3 font-weight-medium"
            href="/Register"
            id="btnRegister"
            onClick={[Function]}
          >
            <b>
              Register
            </b>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="py-2 text-center"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <p
            className=""
            id="dividerOR"
          >
            <strike>
                                  
            </strike>
                
            <b
              className="small font-weight-medium"
            >
              OR
            </b>
                
            <strike>
                                  
            </strike>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="p-0"
  >
    <div
      className="text-center container"
    >
      <div
        className="row text-center"
      >
        <div
          className="col-2-pr-5 col-sm-2 col-md-6 col-lg-2 col-xl-2 ml-auto"
        >
          <img
            alt="fbLogin"
            className="img-fluid d-block mx-auto"
            id="btnFacebookLogin"
            src="fb_icon.png"
          />
        </div>
        <div
          className="col-2-pl-5 col-sm-2 col-md-6 col-lg-2 col-xl-2 mr-auto"
        >
          <img
            alt="googleLogin"
            className="img-fluid d-block mx-auto"
            id="btnGoogleLogin"
            src="google_icon.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
